eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -602,6 +602,65 @@ public function setStatus($contentId, $version, $status)
         return (bool)$statement->rowCount();
     }
 
+    public function setPublishedStatus(int $contentId, int $versionNo): void
+    {
+        $query = $this->getSetVersionStatusQuery(
+            $contentId,
+            $versionNo,
+            VersionInfo::STATUS_PUBLISHED
+        );
+
+        $notExistPublishedVersion = <<< HEREDOC
```

Comment:
Maybe add some inline doc here to explain what this does to go easy on us all in the future, as it contains multiple levels of SQL logic it's not easily readable what it intends to do 😉